eta/core/datasets.py

```diff
@@ -1076,30 +1159,52 @@ def __init__(self, video_path, labels_path, start_frame=None,
                                                   self.duration)
         self.labels_cls = etav.VideoLabels
 
+    def _set_video_metadata(self):
+        video_metadata = etav.VideoMetadata.build_for(self.data_path)
+        self.total_frame_count = video_metadata.total_frame_count
+        self.duration = video_metadata.duration
+        self.end_frame = video_metadata.total_frame_count
 
-class BuilderDataset(object):
+    @classmethod
+    def optional(cls):
+        attrs = super(BuilderVideoRecord, cls).optional()
+        attrs += [
+            "start_frame",
+            "end_frame",
+            "duration",
+            "total_frame_count"
+        ]
+        return attrs
+
+
+class BuilderDataset(etad.DataRecords):
```

Comment:
Note that this is a `DataRecords` instance which are meant to hold `BaseDataRecord`s. `add` and much more included!